samples: power: add check for PM debug hooks, refactor power_mgr sample

nniranjhana opened the pull request:
Due to not checking for the right regex on console, the test was running into errors during automation. Fix it to pass without changing test behavior.

Add the check for pm_debug hooks (which was added by #10655) and enable configs and logging to see the output.

Add the check for lock/unlock PM states (which was added by #11083) and enable corresponding config.

Signed-off-by: Niranjhana N niranjhana.n@intel.com

nniranjhana commented on samples/subsys/power/power_mgr/src/main.c:
```diff
@@ -40,7 +40,7 @@ void main(void)
 	printk(DEMO_DESCRIPTION);
 
-	for (int i = 1; i <= 4; i++) {
+	for (int i = 1; i <= 2; i++) {
```
Well we are using only two cases in the loop - odd and even so I changed it for clarity, it doesn't change the behavior, but if this was there for a reason, I'll change it back

nniranjhana replied:
Will change this back

nniranjhana commented on samples/subsys/power/power_mgr/src/main.c:
```diff
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <device.h>
 #include <gpio.h>
+#include "../subsys/power/policy/pm_policy.h"
```
Okay, done

nniranjhana commented:
@nashif @andrewboie request to review, thanks

nashif commented on samples/subsys/power/power_mgr/CMakeLists.txt:
```diff
@@ -4,3 +4,4 @@ project(power_mgr)
 
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
+target_include_directories(app PRIVATE $ENV{ZEPHYR_BASE}/subsys/power)
```
what exactly do you need from here? We should not include non-public APIs in samples.

nniranjhana replied:
I have removed these, now that Rama's #11656 is merged

nashif commented on samples/subsys/power/power_mgr/src/main.c:
```diff
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <device.h>
 #include <gpio.h>
+#include "policy/pm_policy.h"
```
hmm, is this a private API?

ramakrishnapallala replied:
yes @nashif 

I will move this to include/power.h in my upcoming PR.

nashif replied:
can you submit that PR first?

pizi-nordic commented on samples/subsys/power/power_mgr/src/main.c:
```diff
@@ -56,5 +57,7 @@ void main(void)
 		}
 	}
 
+	sys_pm_dump_debug_info();
```
Do we have any automated way to check if the tests works as expected?

ramakrishnapallala replied:
as of now no, we might have to add new API or expend the existing one to return the details.

Commit:
samples: power: refactor power_mgr sample

Due to not checking for the right regex on console,
the test was running into error during automation.

Fix to get it passing on our infrastructure.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
samples: power: add check for pm_debug hook

Add the check for pm_debug hooks and enable
configs and logging to see the output.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
samples: power: add check for lock/unlock PM states

Enable PM state locking capability and check debug
log counts after sleep with enabling and disabling
entry to low power state.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

nashif commented:
obsolete sample
